test_suite/conftest.py

```diff
@@ -7,6 +7,28 @@
 from lib import test_lib
 
 
+@pytest.fixture(autouse=True)
+def check_run_on(host, request):
+    # This fixture checks the list of combinations of distro-version
+    # each test has. If the host doesn't meet any of the combinations,
+    # the test will be skipped
+    fedora_versions = ['f34', 'f35', 'f36']
+    rhel_versions = ['rhel8.5', 'rhel8.6', 'rhel9.0']
+    centos_versions = ['cs8', 'cs9']
+
+    distro_version_list = [m.args for m in request.node.own_markers if m.name == 'run_on'][0][0]
```

Comment:
This basically gets the list of version that we specify on the `run_on` marker on each test. The `[0][0]` comes from the list comprehension and how the `own_markers` attribute is set up.

Example of what it would look like:

`distro_version_list: ["rhel", "f35", "f36", "cs8"]`